models/ed/R/write.configs.ed.R

```diff
@@ -211,77 +229,63 @@ write.config.ED2 <- function(trait.values, settings, run.id, defaults = settings
-  ed2in.text <- gsub("@START_MONTH@", format(startdate, "%m"), ed2in.text)
-  ed2in.text <- gsub("@START_DAY@", format(startdate, "%d"), ed2in.text)
-  ed2in.text <- gsub("@START_YEAR@", format(startdate, "%Y"), ed2in.text)
-  ed2in.text <- gsub("@END_MONTH@", format(enddate, "%m"), ed2in.text)
-  ed2in.text <- gsub("@END_DAY@", format(enddate, "%d"), ed2in.text)
-  ed2in.text <- gsub("@END_YEAR@", format(enddate, "%Y"), ed2in.text)
-  
-  ##-----------------------------------------------------------------------
-  # Set The flag for IMETAVG telling ED what to do given how input radiation was originally
-  # averaged
-  # -1 = I don't know, use linear interpolation
-  # 0 = No average, the values are instantaneous
-  # 1 = Averages ending at the reference time
-  # 2 = Averages beginning at the reference time
-  # 3 = Averages centered at the reference time Deafult is -1
-  
-  ed2in.text <- gsub("@MET_SOURCE@", -1, ed2in.text)
+
+  thsum <- settings$run$inputs$thsum$path
+  if (!grepl("/$", settings$un$inputs$thsum$path)) {
+    thsum <- paste0(thsum, "/")
+  }
+
+  ed2in.text <- modify_ed2in(
+    ed2in.text,
+    VEG_DATABASE = settings$run$inputs$veg$path,
+    SOIL_DATABASE = settings$run$inputs$soil$path,
+    LU_DATABASE = settings$run$inputs$lu$path,
+    THSUMS_DATABASE = thsum
+  )
   
   ##----------------------------------------------------------------------
   if (is.null(settings$host$scratchdir)) {
     modeloutdir <- file.path(settings$host$outdir, run.id)
   } else {
     modeloutdir <- file.path(settings$host$scratchdir, settings$workflow$id, run.id)
   }
-  ed2in.text <- gsub("@OUTDIR@", modeloutdir, ed2in.text)
-  ed2in.text <- gsub("@ENSNAME@", run.id, ed2in.text)
-  ed2in.text <- gsub("@CONFIGFILE@", file.path(settings$host$rundir, run.id, "config.xml"), ed2in.text)
-  # ed2in.text <- gsub('@CONFIGFILE@','config.xml', ed2in.text) # for ED2.r81 on Kang.  Temporary
-  # hack
+  ed2in.text <- modify_ed2in(
+    ed2in.text,
+    run_name = paste0("ED2 v", revision, " PEcAn ", run.id),
+    run_dir = file.path(settings$host$rundir, run.id),    # For `config.xml`
+    output_dir = modeloutdir    # Sets analysis and history paths
+  )
   
-  ##----------------------------------------------------------------------
-  ed2in.text <- gsub("@FFILOUT@", file.path(modeloutdir, "analysis"), ed2in.text)
-  ed2in.text <- gsub("@SFILOUT@", file.path(modeloutdir, "history"), ed2in.text)
-
   ##---------------------------------------------------------------------
   # Modify any additional tags provided in settings$model$ed2in
-  ed2in.text <- ed2in_set_value_list(settings$model$ed2in, ed2in.text, 
-                                     "PEcAn: Custom argument from pecan.xml")
+  ed2in.text <- modify_ed2in(ed2in.text, .dots = settings$model$ed2in)
   
   ##----------------------------------------------------------------------
-  writeLines(ed2in.text, con = file.path(settings$rundir, run.id, "ED2IN"))
+  check_ed2in(ed2in.text)
```

Comment:
So I guess you created these sets of functions to work locally. And write.configs work locally, but ED2IN paths could be remote paths, right? so I assume this function will give errors because it tries to read remote files for some of the checks, maybe pass a flag for remote and skip those checks?

Reply:
Good point, I didn't consider that. I'll add a `check` argument to all of these functions, and then they can be called with something like `met2model.ED2(..., check = !is_remote)`.

Reply:
Addressed in d463b1f.